merlin/models/tf/core/encoder.py

```diff
@@ -0,0 +1,120 @@
+from typing import Optional, Union
+
+import tensorflow as tf
+from packaging import version
+
+from merlin.models.tf.core import combinators
+from merlin.models.tf.inputs.base import InputBlockV2
+from merlin.models.tf.utils import tf_utils
+from merlin.schema import Schema
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class EncoderBlock(tf.keras.Model):
+    """Block that can be used for prediction but not for train/test"""
+
+    def __init__(
+        self,
+        inputs: Union[Schema, tf.keras.layers.Layer],
+        *blocks: tf.keras.layers.Layer,
+        pre: Optional[tf.keras.layers.Layer] = None,
+        post: Optional[tf.keras.layers.Layer] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if isinstance(inputs, Schema):
+            input_block = InputBlockV2(inputs)
+            self._schema = inputs
+        else:
+            input_block = inputs
+            if not hasattr(inputs, "schema"):
+                raise ValueError("inputs must have a schema")
+            self._schema = inputs.schema
+
+        self.blocks = [input_block] + list(blocks) if blocks else [input_block]
+        self.pre = pre
+        self.post = post
+
+    def call(self, inputs, **kwargs):
+        return combinators.call_sequentially(list(self.to_call), inputs=inputs, **kwargs)
+
+    def build(self, input_shape):
+        combinators.build_sequentially(self, list(self.to_call), input_shape=input_shape)
+        if not hasattr(self.build, "_is_default"):
+            self._build_input_shape = input_shape
+
+    def compute_output_shape(self, input_shape):
+        return combinators.compute_output_shape_sequentially(list(self.to_call), input_shape)
+
+    def __call__(self, inputs, **kwargs):
+        if "features" in kwargs:
+            kwargs.pop("features")
```

Comment:
why do we need to remove the `features`?

Reply:
I will add some docs here. It’s because when we serve this model we won’t get the features param. If we leave it in the model that gets exported expects it.